pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java

```diff
@@ -187,10 +188,38 @@ public void setTimeFormat(String timeFormat) {
     this._timeFormat = timeFormat;
   }
 
+  @Deprecated
+  /**
+   * To check the date format, use
+   *
+   * {@link #isSimpleDateFormat}
+   * {@link #isEpochTimeFormat}
+   *
+   * To get Simple Date format, use {@link #getSimpleDateTimeFormat}
+   *
+   */
   public String getTimeFormat() {
     return _timeFormat;
   }
 
+  public boolean isSimpleDateFormat() {
+    return _timeFormat.startsWith(TimeFormat.SIMPLE_DATE_FORMAT.toString());
+  }
+
+  public boolean isEpochTimeFormat() {
+    return _timeFormat.equals(TimeFormat.EPOCH.toString());
+  }
+
+  public SimpleDateFormat getSimpleDateTimeFormat() {
+    Preconditions.checkArgument(_timeFormat.startsWith(TimeFormat.SIMPLE_DATE_FORMAT.toString()),
+        "Not simple date format");
+    String simpleDateFormatStr = _timeFormat.split(":")[1];
```

Comment:
Simple date format can have colons in it (eg: M/d/yyyy h:mm:ss). In such a case _timeFormat.split(":")[1] will extract only part of the format. Consider using split(separator, limit)?